### PR TITLE
Add Promtail Limit Stage to Grafana Agent Operator

### LINKED ISSUE #2668: Implement promtail stage limits in the Operator
Promtail allows creating [stage limits](https://grafana.com/docs/loki/latest/clients/promtail/stages/limit/). This is be useful to prevent undesired server-side rate limits due to a noisy neighbor or accidental changes, which may potentially lead to log loss in extreme cases.

However, this functionality is not implemented in the Operator (see [CRD template](/home/jorge/workspace/src/github.com/grafana/agent/pkg/operator/config/templates/component/logs/stages.libsonnet)).

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Adds the [limit stage](https://grafana.com/docs/loki/latest/clients/promtail/stages/limit/) to the operator along with generated documentation and code.

#### Which issue(s) this PR fixes
Fixes #2668 

#### PR Checklist

- [X] CHANGELOG updated
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

**Review by marctc — APPROVED** [review]

File: docs/sources/operator/api.md [hunk 1]
```diff
@@ -1770,6 +1770,61 @@ so they get passed to the JMESPath parser.</p>
...
+<code>drop</code><br/>
+<em>
+bool
+</em>
+</td>
+<td>
+<p>When drop is true, log lines that exceed the current rate limit will be discarded.
+When drop is false, log lines that exceed the current rate limit will only wait
+to enter the back pressure mode.</p>
+<p>Defaults to false.</p>
```
Comment (karengermond) [inline docs/sources/operator/api.md:1823]:
```suggestion
<p>When drop is true, log lines that exceed the current rate limit are discarded.
When drop is false, log lines that exceed the current rate limit wait
to enter the back pressure mode.</p>
<p>Defaults to false.</p>
```

**Review by karengermond — APPROVED** [review]
LGTM

**Comment by rfratto** [comment]
@marctc Misha doesn't have merge permissions yet, so one of you or Craig has to merge his PRs :)

**Comment by rfratto** [comment]
(but I'll do it this time since I'm here)

**Comment by rfratto** [comment]
Ah wait, no I won't, Karen's docs feedback haven't been addressed yet (cc @spartan0x117). Once you've addressed it, you can ask Craig, Marc, or I to merge.

**Comment by spartan0x117** [comment]
I pulled the comments from the [loki docs](https://grafana.com/docs/loki/latest/clients/promtail/stages/limit/), should we update that documentation as well, separately?

Also, I committed @karengermond's suggestions, so should be good to merge now :smile:

**Comment by rfratto** [comment]
> Also, I committed @karengermond's suggestions, so should be good to merge now 😄

`api.md` is a generated file, so the changes will need to be made in the CRDs and to regenerate those to rebuild the docs.

**Comment by spartan0x117** [comment]
Ah yeah, that's what I get for trying to do it quickly during lunch, apologies

**Pull request merged**